### PR TITLE
fix cross account issue

### PR DESCRIPTION
**Pull request by vincentni**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/goharbor/harbor/patches/0004-fix-cross-account-issue-for-AWS-ECR.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+diff --git a/src/pkg/reg/adapter/awsecr/adapter.go b/src/pkg/reg/adapter/awsecr/adapter.go
+index 1ad6ea11c..fdc8fae01 100644
+--- a/src/pkg/reg/adapter/awsecr/adapter.go
++++ b/src/pkg/reg/adapter/awsecr/adapter.go
+@@ -49,7 +49,7 @@ func newAdapter(registry *model.Registry) (*adapter, error) {
+ 		return nil, err
+ 	}
+ 	svc, err := getAwsSvc(
+-		region, registry.Credential.AccessKey, registry.Credential.AccessSecret, registry.Insecure, nil)
++		region, registry.Credential.AccessKey, registry.Credential.AccessSecret, registry.Insecure, &registry.URL)
```
Comment (junshun) [inline projects/goharbor/harbor/patches/0004-fix-cross-account-issue-for-AWS-ECR.patch:19]: Isnt there another change needed to also use this registry.URL to pass into registry ids for the getAuthorizationToken function call

Comment (vincentni) [reply projects/goharbor/harbor/patches/0004-fix-cross-account-issue-for-AWS-ECR.patch:19]: With that change, [the following lines](https://github.com/goharbor/harbor/blob/main/src/pkg/reg/adapter/awsecr/auth.go#L116-L118) will be executed so the endpoint is explicitly set there. I tested it and it is working now.

Comment (junshun) [reply projects/goharbor/harbor/patches/0004-fix-cross-account-issue-for-AWS-ECR.patch:19]: Nice

**Comment by junshun** [comment]
https://github.com/goharbor/harbor/issues/15388

Putting upstream issue here to keep track of.
We should get this change upstream if it works.

**Comment by vincentni** [comment]
https://github.com/goharbor/harbor/pull/17533

The change is also upstreamed.

**Pull request merged**